File: buddha-hr-backend/buddha-hr-backend/src/main/java/com/buddhahr/job/service/JobService.java
package com.buddhahr.job.service;

import com.buddhahr.job.EmploymentType;
import com.buddhahr.job.dto.request.CreateJobRequest;
import com.buddhahr.job.dto.response.JobResponse;
import com.buddhahr.job.entity.JobCategory;
import org.springframework.data.domain.Page;

import java.util.List;

public interface JobService {

    JobResponse createJob(CreateJobRequest request);

    Page<JobResponse> getAllJobs(
            int page,
            int size,
            String sortBy,
            String direction
    );

    JobResponse getJobById(Long id);

    JobResponse updateJob(Long id, CreateJobRequest request);

  Page<JobResponse> searchJobs(
        String keyword,
        JobCategory category,
        String location,
        EmploymentType employmentType,
        int page,
        int size
 );
    Page<JobResponse> getAllJobsForAdmin(
            int page,
            int size,
            String sortBy,
            String direction
    );

    void archiveJob(Long id);
    void restoreJob(Long id);

}

File: Buddha-Hr-frontend/buddha-hr-frontend/src/services/jobService.js
import api from "./api";

/**
 * Get all jobs with optional filters
 */
export const getJobs = async (
  page = 0,
  size = 9,
  filters = {},
  sortBy = "createdAt",
  direction = "desc"
) => {

  const response = await api.get("/jobs", {
    params: {
      page,
      size,
      sortBy,
      direction,

      keyword: filters.keyword || undefined,
      category: filters.category || undefined,
      location: filters.location || undefined,
      employmentType: filters.type || undefined,
    },
  });

  return response.data;
};

/**
 * Get single job
 */
export const getJob = async (id) => {

  const response = await api.get(`/jobs/${id}`);

  return response.data;
};

/**
 * Create job
 */
export const createJob = async (job) => {

  const response = await api.post("/jobs", job);

  return response.data;
};

/**
 * Update job
 */
export const updateJob = async (id, job) => {

  const response = await api.put(`/jobs/${id}`, job);

  return response.data;
};

/**
 * Archive (disable) job
 */
export const archiveJob = async (id) => {

  return api.patch(`/jobs/${id}/archive`);

};

/**
 * Restore (enable) job
 */
export const restoreJob = async (id) => {

  return api.patch(`/jobs/${id}/restore`);

};

/**
 * Delete job (if still used)
 */
export const deleteJob = async (id) => {

  return api.delete(`/jobs/${id}`);

};

export const getAdminJobs = async (
    page = 0,
    size = 10,
    sortBy = "createdAt",
    direction = "desc"
) => {

    const response = await api.get("/jobs/admin", {
        params: {
            page,
            size,
            sortBy,
            direction,
        },
    });

    return response.data;
};